miio/airconditioner_miot.py

```diff
@@ -0,0 +1,403 @@
+            {
+                prop["did"]: prop["value"] if prop["code"] == 0 else None
+                for prop in self.get_properties_for_mapping()
+            }
+        )
+
+    @command(default_output=format_output("Powering on"))
+    def on(self):
+        """Power on."""
+        return self.set_property("power", True)
+
+    @command(default_output=format_output("Powering off"))
+    def off(self):
+        """Power off."""
+        return self.set_property("power", False)
+
+    @command(
+        click.argument("mode", type=EnumType(OperationMode, False)),
+        default_output=format_output("Setting operation mode to '{mode.value}'"),
+    )
+    def set_mode(self, mode: OperationMode):
+        """Set operation mode."""
+        return self.set_property("mode", mode.value)
+
+    @command(
+        click.argument("target_temperature", type=float),
+        default_output=format_output(
+            "Setting target temperature to {target_temperature}"
+        ),
+    )
+    def set_target_temperature(self, target_temperature: float):
+        """Set target temperature in Celsius."""
+        if (
+            target_temperature < 16.0
+            or target_temperature > 31.0
+            or target_temperature % 0.5 != 0
+        ):
+            raise AirConditionerMiotException(
+                "Invalid target temperature: %s" % target_temperature
+            )
+        return self.set_property("target_temperature", target_temperature)
+
+    @command(
+        click.argument("eco", type=bool),
+        default_output=format_output(
+            lambda eco: "Turning on ECO mode" if eco else "Turning off ECO mode"
+        ),
+    )
+    def set_eco(self, eco: bool):
+        """Turn ECO mode on/off."""
+        return self.set_property("eco", eco)
+
+    @command(
+        click.argument("heater", type=bool),
+        default_output=format_output(
+            lambda heater: "Turning on heater" if heater else "Turning off heater"
+        ),
+    )
+    def set_heater(self, heater: bool):
+        """Turn aux heater mode on/off."""
+        return self.set_property("heater", heater)
+
+    @command(
+        click.argument("dryer", type=bool),
+        default_output=format_output(
+            lambda dryer: "Turning on dryer" if dryer else "Turning off dryer"
+        ),
+    )
+    def set_dryer(self, dryer: bool):
+        """Turn aux dryer mode on/off."""
+        return self.set_property("dryer", dryer)
+
+    @command(
+        click.argument("sleep_mode", type=bool),
+        default_output=format_output(
+            lambda sleep_mode: "Turning on sleep mode"
+            if sleep_mode
+            else "Turning off sleep mode"
+        ),
+    )
+    def set_sleep_mode(self, sleep_mode: bool):
+        """Turn sleep mode on/off."""
+        return self.set_property("sleep_mode", sleep_mode)
+
+    @command(
+        click.argument("fan_level", type=EnumType(FanLevel, False)),
```

Comment:
```suggestion
        click.argument("fan_level", type=EnumType(FanLevel)),
```
`EnumType` defaults to incasesensitive since some time, so there is no need to pass False here.